### PR TITLE
[DX-1322] [TT-11955] - operator release 0.17.1

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

[DX-1322](https://tyktech.atlassian.net/browse/DX-1322)

---

### Preview Link

<br>

[Preview](https://deploy-preview-4603--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/release-notes/operator-0.17/)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

___

## **Type**
enhancement, documentation

___

## **Description**
- Added detailed release notes for Tyk Operator version 0.17.1 in the existing documentation.
- The new release notes include:
  - No breaking changes and deprecations.
  - Upgrade instructions linked to the official upgrading guide.
  - Download links for Docker images and source code.
  - A changelog detailing bug fixes, including security fixes and improvements in handling certificates with ingress.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>operator-0.17.md</strong><dd><code>Add Release Notes for Tyk Operator Version 0.17.1</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md
<li>Added a new section for Tyk Operator release 0.17.1.<br> <li> Provided detailed release notes including upgrade instructions, <br>download links, and a changelog.<br> <li> Highlighted that the release focuses on bug fixes.<br> <li> Included specific bug fixes and improvements in the changelog.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4603/files#diff-a365522c7ee19f47d161d3de7faf0804b8afa6a93f93e43cd3748105251df31e">+51/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

[DX-1322]: https://tyktech.atlassian.net/browse/DX-1322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md [hunk 1]
```diff
@@ -10,6 +10,57 @@ description: "Release notes documenting updates, enhancements, fixes and changes
...
+<li>
+<details>
+<summary>Reference helm values in webhook and controller manager services </summary>
+
+</details>
+</li>
+
+<li>
+<details>
+<summary>Fixed: CVE </summary>
```
Comment (buraksekili) [inline tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md]: we fixed two CVEs: CVE-2023-45288 and CVE-2024-24786
```suggestion
<summary>Fixed CVE-2023-45288 and CVE-2024-24786 </summary>
```

**Review by buraksekili — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md [hunk 2]
```diff
@@ -10,6 +10,68 @@ description: "Release notes documenting updates, enhancements, fixes and changes
...
+##### Release date 6 May 2024
+
+#### Breaking Changes
+This release has no breaking changes.
+
+#### Deprecations
+There are no deprecations in this release.
+
+#### Upgrade Instructions
+Go to the [Upgrading Tyk Operator]({{<ref "tyk-stack/tyk-operator/installing-tyk-operator#upgrading-tyk-operator">}}) section for detailed upgrade Instructions.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md]: Update to make instructions lower case

File: tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md [hunk 3]
```diff
@@ -10,6 +10,68 @@ description: "Release notes documenting updates, enhancements, fixes and changes
...
+- Source code tarball - [Tyk Operator Repo](https://github.com/TykTechnologies/tyk-operator/releases/tag/v0.17.1)
+
+#### Changelog {#Changelog-v0.17.1}
+
+##### Fixed
+
+<ul>
+<li>
+<details>
+<summary>Fixed ApiDefinition CRs generated by the Ingress Controller used a wrong certiciate</summary>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md]: update typo for certificate and expand CR abbreviation

File: tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md [hunk 4]
```diff
@@ -10,6 +10,68 @@ description: "Release notes documenting updates, enhancements, fixes and changes
...
+
+When using Tyk as Ingress Controller with TLS enabled, the ApiDefinition CRs generated by the Ingress Controller is missing OrgID field. As a result, Tyk Gateway used a wrong certificate when serving a request. It is fixed by adding back OrgID field to ApiDefinition CRs created by Ingress Controller.</summary>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md]: Expand CR abbreviation

```suggestion
When using Tyk as an Ingress Controller with TLS enabled, the ApiDefinition CRs generated by the Ingress Controller is missing the OrgID field. As a result, Tyk Gateway used a wrong certificate when serving a request. It is fixed by adding back OrgID field to ApiDefinition CRs created by Ingress Controller.</summary>
```

File: tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md [hunk 5]
```diff
@@ -10,6 +10,68 @@ description: "Release notes documenting updates, enhancements, fixes and changes
...
+<details>
+<summary>Addressed security vulnerabilities CVE-2023-45288</summary>
+
+Addressed security vulnerabilities [CVE-2023-45288](https://nvd.nist.gov/vuln/detail/CVE-2023-45288) where an attacker may cause an HTTP/2 endpoint to read arbitrary amounts of header data by sending an excessive number of CONTINUATION frames. Maintaining HPACK state requires parsing and processing all HEADERS and CONTINUATION frames on a connection. When a request's headers exceed MaxHeaderBytes, no memory is allocated to store the excess headers, but they are still parsed. This permits an attacker to cause an HTTP/2 endpoint to read arbitrary amounts of header data, all associated with a request which is going to be rejected. These headers can include Huffman-encoded data which is significantly more expensive for the receiver to decode than for an attacker to send. The fix sets a limit on the amount of excess header frames we will process before closing a connection.
+</details>
+</li>
+
+<li>
+<details>
+<summary>Addressed security vulnerabilities CVE-2024-24786 </summary>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md]: remove extra space before ending summary tag

File: tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md [hunk 6]
```diff
@@ -10,6 +10,68 @@ description: "Release notes documenting updates, enhancements, fixes and changes
...
+
+Addressed security vulnerabilities [CVE-2024-24786](https://nvd.nist.gov/vuln/detail/CVE-2024-24786) where the protojson.Unmarshal function can enter an infinite loop when unmarshaling certain forms of invalid JSON. This condition can occur when unmarshaling into a message which contains a google.protobuf.Any value, or when the UnmarshalOptions.DiscardUnknown option is set.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-operator/release-notes/operator-0.17.md]: Add backticks, e.g. protojson.Unmarshal etc.

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Thanks for submitting release notes for operator 0.17.1 @caroltyk I have made minor suggestions before merge and release. Is this targetted for release 5.3.0 only?

**Comment by carolwkc** [comment]
Thanks for the review @dcs3spp . Docs updated as per comments.

**Review by dcs3spp — APPROVED** [review]
Thanks @caroltyk PR LGTM, I'll merge and release to 5.3.0

**Pull request merged**